packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -1307,15 +1329,23 @@ class _CupertinoActionSheetActionState extends State<CupertinoActionSheetAction>
 
 // Renders the background of a button (both the pressed background and the idle
 // background) and reports its state to the parent with `onPressStateChange`.
+//
+// Although this class doesn't keep any states, it's still a stateful widget
+// because the state is used as a persistent object across rebuilds to provide
+// to [MetaData.data].
```

Comment:
That's a great idea!

Reply:
On a second thought, I found it harder than imagined because the `_CupertinoActionSheetState` has to keep track of the number of children and resize the list of `_SlideTarget` objects based on that, which is not the most cleanest code. I would just rather keep the stateful widget.

Reply:
sounds good